modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AliasAnnotationParser.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.painless.spi.annotation;
+
+import java.util.Map;
+
+public class AliasAnnotationParser implements WhitelistAnnotationParser {
+    public static final AliasAnnotationParser INSTANCE = new AliasAnnotationParser();
+
+    public AliasAnnotationParser() {}
+
+    @Override
+    public Object parse(Map<String, String> arguments) {
+        if (arguments.size() != 1) {
+            throw new IllegalArgumentException("[@alias] requires one alias");
+        }
+        AliasAnnotation annotation = null;
+        for (Map.Entry<String, String> entry : arguments.entrySet()) {
+            if ("class".equals(entry.getKey()) == false) {
+                throw new IllegalArgumentException("[@alias] only supports class aliases");
+            }
+            String alias = entry.getValue();
+            if (alias == null || "".equals(alias)) {
```

Comment:
nit: use `alias.isBlank()` for checking empty

Reply:
Changed.